### ISSUE TITLE
@JiaRenChang

### ISSUE THREAD

**Issue by chenweiweiwei** [JiaRenChang/PSMNet#75]
how can I get the disparity error?

**Comment by aasharma90** [comment]
Simply follow the test() function in finetune.py. It returns the associated 3px (default for KITTI) disparity error for your predicted disparity output.

**Comment by chenweiweiwei** [comment]
thanks for your guide , I have another question , which is about how to set the validation set ? thanks !

**Comment by JiaRenChang** [comment]
Also follow the finetune.py.
You can check that the KITTI dataset has been split into training and valiation set in  finetune.py.

**Comment by chenweiweiwei** [comment]
parser.add_argument('--seed', type=int, default=1, metavar='S',
                    help='random seed (default: 1)')
Should I modify this?

**Comment by chenweiweiwei** [comment]
so, where should I modify? can you copy it to me ? thanks!

**Comment by aasharma90** [comment]
@chenweiweiwei 
This is about your question related to "how to set the validation set?"

If you follow KITTIloader2015.py (called within finetune.py, if you are fine-tuning on KITTI2015), you can see that the author has distributed the entire dataset containing 200 image pairs into 'train' (with the first 160 pairs), and 'val' (with the remaining 40 pairs) sets respectively. He is doing this because there is no explicit validation set for KITTI2015. 

You can either do the same for your own dataset (i.e. divide it into 'train' and 'val' say in the ratio of 4:1), or if you have an explicit validation set available, use it completely for 'val'. 

Hope it helps!

**Comment by JiaRenChang** [comment]
Thanks for the clarification, @aasharma90.

**Comment by chenweiweiwei** [comment]
@aasharma90  thanks for your help !

**Comment by ycloving1314** [comment]
why do I use the devkit_scene_flow to calculate disparity error ,my disparity error is 100%? is my disparity map is wrong?

**Comment by aasharma90** [comment]
Hi @ycloving1314 

I am not sure what you mean by 'why do I use ...'; if I assume you mean 'how do I use ...', and you say your results are 100% incorrect, then you should probably check the scale of `D_gt` and `D_est` inputs going to the `disp_error()` function provided in the dev_kit. 

ps: I also evaluate the disparity maps using the dev_kit, and I haven't encountered any problems so far. You can reach out to me at aashish.sharma@u.nus.edu if you need more clarification. 

Hope it helps!
